website_version_ce/models/google_management.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-#
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from datetime import datetime, timedelta
+import logging
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from openerp.exceptions import Warning as UserError
+from openerp import models, exceptions, api, fields
+from openerp.tools.translate import _
+import simplejson
+
+_logger = logging.getLogger(__name__)
+
+
+class GoogleManagement(models.AbstractModel):
+
+    STR_SERVICE = 'management'
+
+    _name = 'google.%s' % STR_SERVICE
+
+    @api.model
+    def generate_data(self, experiment):
+        return {
+            'name': experiment['name'],
+            'status': experiment['status'],
+            'variations': experiment['variations']
+        }
+
+    @api.model
+    def _setup_exp_variables(self, website_id, check=False):
+        gs_pool = self.env['google.service']
+        website = self.env['website'].browse(website_id)
+        web_property_id = website.google_analytics_key
+        action_id = self.env['ir.model.data'].xmlid_to_res_id(
+            'website_version_ce.action_website_view')
+
+        if check and not web_property_id:
+            raise exceptions.RedirectWarning(_(
+                'Click on the website you want to make A/B testing and '
+                'configure the Google Analytics Key and View ID'),
+                action_id,
+                _('go to the websites menu')
+            )
+
+        account_id = web_property_id.split('-')[1]
+        profile_id = website.google_analytics_view_id
+
+        return (gs_pool, website, web_property_id,
+                action_id, account_id, profile_id)
+
+    @api.model
+    def create_an_experiment(self, data, website_id):
+        (gs_pool, website, web_property_id, action_id, account_id,
+         profile_id) = self._setup_exp_variables(website_id, True)
```

Comment:
Ummm the module seems broken in a lot of ways right now honestly, but once it compiles I'll give it a shot. Thanks!